platform-sdk/swirlds-common/src/main/java/com/swirlds/common/metrics/PlatformMetricsFactory.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2022-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.metrics;
+
+import java.util.Objects;
+
+/**
+ * Factory for all {@link Metric}-implementations
+ */
+public interface PlatformMetricsFactory extends MetricsFactory {
```

Comment:
Please add nullity annotations to all function parameters and return types in this class.